### PR TITLE
GSoC'22 Final Report: Improving the Calendar Application

### PR DESCRIPTION
**Pull request by harshit-sharma-gits**
Final report for the GSoC Project: Improving the Calendar Application

### CODE REVIEW

**Comment by pulkomandy** [comment]
Please tag your github repository with the version at the end of GSoC, and include a link to the tag in the blogpost.

This way, Google knows exactly what was funded during GSoC, even if you make more changes to the code later.

@nielx @humdingerb anything else to add to the report?

File: content/blog/harshitsharma/2022-09-12_gsoc_2022_final_report_improving_the_calendar_application.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+| https://github.com/HaikuArchives/Calendar/pull/124 | Code restructure for Daemon. Restructured the files, so there is a Calendar app and a separate Daemon |
+| https://github.com/HaikuArchives/Calendar/pull/125 | Removed dependence of the DBManager on the Main Application |
+| https://github.com/HaikuArchives/Calendar/pull/126 | Set up a basic daemon which node monitors the Events directory for events adding/removing. |
+| https://github.com/HaikuArchives/Calendar/pull/127 | UI change in Calendar and data model changes |
+| https://github.com/HaikuArchives/Calendar/pull/128 | Querying the DB for an updated list of events |
+| https://github.com/HaikuArchives/Calendar/pull/129 | Getting alerts at the right time using `BMessageRunner` |
+
+## What's next?
+
+The work for Daemon is (almost) complete.
```
Comment (nielx) [inline content/blog/harshitsharma/2022-09-12_gsoc_2022_final_report_improving_the_calendar_application.md:65]: Please elaborate on open ends. If necessary, add issues for them in the Calendar repository and link them here.

File: content/blog/harshitsharma/2022-09-12_gsoc_2022_final_report_improving_the_calendar_application.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+What remains is: Automatically starting the CalendarDaemon with the help of `launch_daemon`. I would definitely complete this.
+
+I've decided to continue to contribute to open source in my spare time. And I'll continue to engage with the Haiku community as well. Projects here are really interesting!
+
+---
+
+# Final Words
+
+I had a great and very productive time working on this project. Thanks to this project, I have learnt many things about Haiku, git, and even C++!
```
Comment (nielx) [inline content/blog/harshitsharma/2022-09-12_gsoc_2022_final_report_improving_the_calendar_application.md:74]: Please elaborate on what you learned, and what could have gone better. I am very interested in learning where you got stuck, what helped you forward. You could also share a list of resources that were useful to you, in order to help future contributors figuring it out.

**Review by nielx — CHANGES_REQUESTED** [review]
I am also missing how you tested your changes, and a link to the tool you used to generate demo data.

**Comment by harshit-sharma-gits** [comment]
> Please tag your github repository with the version at the end of GSoC, and include a link to the tag in the blogpost.

I've already included my fork there, on line 21

**Comment by pulkomandy** [comment]
> > Please tag your github repository with the version at the end of GSoC, and include a link to the tag in the blogpost.
> 
> I've already included my fork there, on line 21

I asked for a tag, not just a link to the github fork. With the link you provided, any work you do in the future on the Calendar app will be shown there. Google needs to know clearly which part was done during GSoC and which part was done after it if that happens. The best way to achieve that is with a git tag, which will remain pointing to the same commit (the last one you did for GSoC) even if you decide to do more work later on the app.

This is important for Google accountants to make sure there is actual work done during GSoC and they are not giving you money for nothing. This is the only proof they will have. So ti's important to follow this rule.

**Comment by harshit-sharma-gits** [comment]
@pulkomandy @nielx

Please check out the blog now. I suppose it fulfils all the above stated requirements now!

**Pull request merged**